assets.php

```diff
@@ -0,0 +1,165 @@
+	if ( empty( $dependency_file ) ) {
+		return [];
+	}
+
+	// Ensure the filepath is valid.
+	if ( ! file_exists( $dependency_file ) || 0 !== validate_file( $dependency_file ) ) {
+		return [];
+	}
+
+	// Try to load the dependencies.
+	// phpcs:ignore WordPressVIPMinimum.Files.IncludingFile.UsingVariable
+	$dependencies = require $dependency_file;
+	if ( empty( $dependencies['dependencies'] ) || ! is_array( $dependencies['dependencies'] ) ) {
+		return [];
+	}
+
+	return $dependencies['dependencies'];
+}
+
+/**
+ * Get the contentHash for a given asset.
+ *
+ * @param string $asset Entry point and asset type separated by a '.'.
+ *
+ * @return string The asset's hash.
+ */
+function get_asset_hash( string $asset ) : string {
+	return get_asset_property( $asset, 'hash' )
+		?? CLONE_REPLACE_ASSET_MAP['hash']
+		?? '1.0.0';
+}
+
+/**
+ * Get the URL for a given asset.
+ *
+ * @param string  $asset Entry point and asset type separated by a '.'.
+ * @param boolean $dir   Optional. Whether to return the directory path or the plugin URL path. Defaults to false (returns URL).
+ *
+ * @return string The asset URL.
+ */
+function get_asset_path( string $asset, bool $dir = false ) : string {
+	// Try to get the relative path.
+	$relative_path = get_asset_property( $asset, 'path' );
+	if ( empty( $relative_path ) ) {
+		return '';
+	}
+
+	// Negotiate the base path.
+	$base_path = true === $dir
+		? dirname( __FILE__ ) . '/build'
+		: plugins_url( 'build', __FILE__ );
+
+	return trailingslashit( $base_path ) . $relative_path;
+}
+
+/**
+ * Get a property for a given asset.
+ *
+ * @param string $asset Entry point and asset type separated by a '.'.
+ * @param string $prop The property to get from the entry object.
+ *
+ * @return string|null The asset property based on entry and type.
+ */
+function get_asset_property( string $asset, string $prop ) : ?string {
+	/*
+	 * Appending a '.' ensures the explode() doesn't generate a notice while
+	 * allowing the variable names to be more readable via list().
+	 */
+	list( $entrypoint, $type ) = explode( '.', "$asset." );
+
+	$asset_property = CLONE_REPLACE_ASSET_MAP[ $entrypoint ][ $type ][ $prop ] ?? null;
+
+	return $asset_property ? $asset_property : null;
+}
+
+/**
+ * Creates a new Jed instance with specified locale data configuration.
+ *
+ * @param string $to_handle The script handle to attach the inline script to.
+ */
+function inline_locale_data( string $to_handle ) {
+	global $post;
+
+	// Define locale data for Jed.
+	$locale_data = [
+		'' => [
+			'domain'  => 'clone-replace',
+			'lang'    => is_admin() ? get_user_locale() : get_locale(),
+		],
+	];
+
+	// Pass the Jed configuration to the admin to properly register i18n.
+	wp_add_inline_script(
+		$to_handle,
+		'wp.i18n.setLocaleData( ' . wp_json_encode( $locale_data ) . ", 'clone-replace' );"
+	);
+
+	$json = [
+		'nonce'   => wp_create_nonce( 'clone_post_' . $post->ID ),
+		'homeUrl' => admin_url(),
```

Comment:
`home_url()` is a valid function, so I'd recommend naming this `adminUrl` in case at some point in the future we'd actually need to pass `home_url()` as well.
```suggestion
		'adminUrl' => admin_url(),
```